runtime/arm64.S

```diff
@@ -285,29 +375,69 @@ FUNCTION(caml_allocN)
 
 FUNCTION(caml_c_call)
         CFI_STARTPROC
-    /* Preserve return address in callee-save register x19 */
-        mov     x19, x30
-        CFI_REGISTER(30, 19)
-    /* Record lowest stack address and return address */
-        str     x30, Caml_state(last_return_address)
-        add     TMP, sp, #0
-        str     TMP, Caml_state(bottom_of_stack)
+        stp     x29, x30, [sp, -16]!
+        CFI_ADJUST(16)
+        add     x29, sp, #0
+    /* Switch from OCaml to C */
+        SWITCH_OCAML_TO_C
     /* Make the exception handler alloc ptr available to the C code */
         str     ALLOC_PTR, Caml_state(young_ptr)
-        str     TRAP_PTR, Caml_state(exception_pointer)
+        str     TRAP_PTR, Caml_state(exn_handler)
     /* Call the function */
         blr     ADDITIONAL_ARG
     /* Reload alloc ptr  */
         ldr     ALLOC_PTR, Caml_state(young_ptr)
+    /* Load ocaml stack */
+        SWITCH_C_TO_OCAML
     /* Return */
-        ret     x19
+        ldp     x29, x30, [sp], 16
+        ret
         CFI_ENDPROC
         END_FUNCTION(caml_c_call)
 
+FUNCTION(caml_c_call_stack_args)
+        CFI_STARTPROC
+    /* Arguments:
+        C arguments  : x0 to x7, d0 to d7
+        C function   : ADDITIONAL_ARG
+        C stack args : begin=STACK_ARG_BEGIN
+                       end=STACK_ARG_END */
+        stp     x29, x30, [sp, -16]!
+        CFI_ADJUST(16)
+        add     x29, sp, #0
+    /* Switch from OCaml to C */
+        SWITCH_OCAML_TO_C
+    /* Make the exception handler alloc ptr available to the C code */
+        str     ALLOC_PTR, Caml_state(young_ptr)
+        str     TRAP_PTR, Caml_state(exn_handler)
+    /* Store sp to restore after call */
+        mov     x19, sp
+    /* Copy arguments from OCaml to C stack
+       NB: STACK_ARG_{BEGIN,END} are 16-byte aligned */
+1:      sub     STACK_ARG_END, STACK_ARG_END, 16
+        cmp     STACK_ARG_END, STACK_ARG_BEGIN
+        b.lo    2f
+        ldp     TMP, TMP2, [STACK_ARG_END]
+        stp     TMP, TMP2, [sp, -16]!; CFI_ADJUST(16)
+        b       1b
+2:  /* Call the function */
+        blr     ADDITIONAL_ARG
+    /* Restore stack */
+        mov     sp, x19
```

Comment:
```suggestion
        mov     sp, x29
```

Reply:
[resolved, see comment on x19 in previous comment around line 420]